app/assets/javascripts/discourse/app/services/users-api.js

```diff
@@ -0,0 +1,13 @@
+import Service from "@ember/service";
+import { ajax } from "discourse/lib/ajax";
+import User from "discourse/models/user";
+
+export default class UsersApiService extends Service {
+  async lookupUsers(usernames) {
+    const response = await ajax(`/u/lookup/users.json`, {
+      type: "GET",
+      data: { usernames },
+    });
+    return response.users.map((u) => User.create(u));
```

Comment:
Using the static `create` method on a model is to be avoided. Use the store service instead

```js
@service store;
```

```suggestion
    return response.users.map((u) => this.store.createRecord("user", u));
```

(btw. note to self and modernize-js team/@davidtaylorhq - we should update the store so it can also accept a class instead of a string 😃)